### PR TITLE
Add:support attachments

### LINKED ISSUE #49: Imp: support attachments
In dubbo protocol, attachments need to be supported. And calling method of hessian will be changed in this improvement. 

- Maybe, it could be applyed to define a Request and Response.

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

  - go mod tidy
  - use `Request` and `Response`

**Which issue(s) this PR fixes**:

Fixes #49

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

  - user will make a Request or Response instead of body content.

### CODE REVIEW

File: hessian.go [hunk 1]
```diff
@@ -175,16 +175,15 @@ func (h *HessianCodec) ReadBody(rspObj interface{}) error {
...
-		}
-		rsp.Exception = ErrJavaException
 		decoder := NewDecoder(buf[:])
 		exception, err := decoder.Decode()
 		if err != nil {
 			return perrors.WithStack(err)
 		}
+		rsp, ok := rspObj.(*Response)
+		if !ok {
+			return perrors.Errorf("java exception:%s", exception.(string))
```
Comment (gaoxinge) [inline hessian.go]: Why not return type assertion error?

File: response.go [hunk 2]
```diff
@@ -28,19 +28,43 @@ import (
...
-	RspObj    interface{}
-	Exception error
-	//Attachments map[string]string
+	RspObj      interface{}
+	Exception   error
+	Attachments map[string]string
+}
+
+// Pls just use NewResponse to get a Response.
+func NewResponse(rspObj interface{}, exp error, atta map[string]string) *Response {
```
Comment (gaoxinge) [inline response.go]: Use `exception` instead of `exp` and `attachment` instead of `atta`.

Comment (fangyincheng) [reply response.go]: ok

File: hessian.go (hunk 1 above)
Comment (fangyincheng) [reply hessian.go]: For compatibility with previous versions.

File: response.go [hunk 3]
```diff
@@ -28,19 +28,43 @@ import (
...
 type Response struct {
-	RspObj    interface{}
-	Exception error
-	//Attachments map[string]string
+	RspObj      interface{}
+	Exception   error
+	Attachments map[string]string
+}
+
+// Pls just use NewResponse to get a Response.
```
Comment (wongoo) [inline response.go]: change the comment to `NewResponse create a new Response`

File: request.go [hunk 4]
```diff
@@ -146,20 +146,43 @@ func getArgsTypeList(args []interface{}) (string, error) {
 	return types, nil
 }
 
+type Request struct {
+	Params      interface{}
+	Attachments map[string]string
+}
+
+// Pls just use NewRequest to get a Request.
```
Comment (wongoo) [inline request.go]: change the comment to `NewRequest create a new request`

**Comment by wongoo** [comment]
@fangyincheng this pr is not forward compatible, I'd like to merge it to a new branch `feature-attachment` so that you can test it in dubbogo. 

@AlexStocks FYI

File: hessian.go (hunk 1 above)
Comment (gaoxinge) [reply hessian.go]: @fangyincheng ok.

File: response.go (hunk 3 above)
Comment (fangyincheng) [reply response.go]: ok

**Comment by fangyincheng** [comment]
> @fangyincheng this pr is not forward compatible, I'd like to merge it to a new branch `feature-attachment` so that you can test it in dubbogo.
> 
> @AlexStocks FYI

It is forward compatible. Compatible with previous usage.

File: request.go (hunk 4 above)
Comment (fangyincheng) [reply request.go]: ok

**Comment by wongoo** [comment]
@fangyincheng @gaoxinge @hxmhlt @u0x01  all conversions in the pr have being resolved, if there is not any other question, I will merge it tomorrow.

**Pull request merged**